spring-integration-azure/spring-integration-azure-core/src/main/java/com/microsoft/azure/spring/integration/core/DefaultMessageHandler.java

```diff
@@ -86,12 +86,16 @@ protected void handleMessageInternal(Message<?> message) throws Exception {
     private void handleSendResponseAsync(Message<?> message, CompletableFuture<?> future) {
         future.handle((t, ex) -> {
             if (ex != null) {
-                log.warn("{} sent failed in async mode", message);
+                if (log.isWarnEnabled()) {
```

Comment:
just curiosity, the `log` cannot handle this logic like `if enabled` ?

Reply:
Without if condition, even if log is disabled, log string still need to be built.

Reply:
I see, you are avoiding the creation of new String instance. My suggestion that may not be one issue, anyway, it is up to your choice.

Reply:
Actually, it's a really big issue. Since this log will be happening for each message, this will fill memory quickly.

Reply:
Compiler has unique String optimization, and you may need to provide profile data about how extra memory quickly.

Reply:
Each message will be different. So this optimization won't help.